static/src/js/vuex/library/actions.js

```diff
@@ -1,20 +1,47 @@
 import api from '../../api';
 import constants from '../../constants';
 import transformTextGroupList from './transforms';
+import utils from './utils';
 
 export default {
-  [constants.LIBRARY_LOAD_TEXT_GROUP_LIST]: ({ commit }) => api.getTextGroupList((data) => {
+  [constants.LIBRARY_LOAD_TEXT_GROUP_LIST]: ({ commit }) => {
+    const currentVersion = process.env.API_VERSION;
+    const version = localStorage.getItem('libraryDataVersion');
+    const libraryData = JSON.parse(localStorage.getItem('libraryData'));
+
+    if (!version || parseInt(version, 10) < parseInt(currentVersion, 10) || !libraryData) {
+      return api.getTextGroupList((data) => {
+        // add data to localStorage
+        try {
+          localStorage.setItem('libraryData', JSON.stringify(data));
+          localStorage.setItem('libraryDataVersion', currentVersion);
+        } catch (e) {
+          if (utils.isQuotaExceeded(e)) {
+            console.error('localStorage is full!'); // eslint-disable-line no-console
```

Comment:
@jacobwegner Thoughts on maybe logging an info message here or something so we can see how many times this happens?

Reply:
Send a quick AJAX message back to the server and creating a log message.

Reply:
@mjhea0 I think console output is fine for now; I think longer-term we'll likely have a more efficient load of the library (faster BE response and perhaps chunking on the FE with or using [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).

I'd rather take the small win than introduce more changes at this time.

Reply:
Yeah, I spiked a bit yesterday on streaming/server-side events via Django's `StreamingHttpResponse`. It is a bit strange how it works, especially with WSGI. Django Channels plus ASGI would be the better approach.

I got it working on the server, but I would have had to do a pretty major refactor on the client to handle it. 

In the end, I don't think streaming is the right way to go. It's a major refactor on both ends. I suggest doing an initial request of like 50 libraries and then requesting the rest after the initial render. Then, in terms of caching, there's less stress on localStorage since you can just store 50 objects rather than like 350.